### PR TITLE
MARKET-1336 Treemap

### PR DESCRIPTION
**Pull request by Saharij**
Screenshot 2022-12-15 at 01 06 14

### CODE REVIEW

File: components/bl-treemap-component/src/index.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { useMemo } from 'react';
+import TreeMap from "./lib/treemap.min.js";
+
+const { cn } = BackendlessUI.CSSUtils;
+
+export default function MyCustomComponent({ component }) {
+  const { classList, display, disabled, label, valueUnit, width, height, data } = component;
```
Comment (StasivPavlo) [inline components/bl-treemap-component/src/index.js:7]: add property `style`

File: components/bl-treemap-component/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Treemap
+Treemap is a component of Backendless UI-Builder designer. The main purpose of the component is proportional display of data. Also, the [react-d3-treemap](https://github.com/jquintozamora/react-d3-treemap) library is used to create this component.
```
Comment (v-excelsior) [inline components/bl-treemap-component/README.md:2]: Please, take a look at this library

image

I don't think it's enough level of code quality to share it with customers which pay for our services.

Please, found the more popular solution with a rich API, or write it by yourself - to have the possibility of delivering new features.

**Review by v-excelsior — CHANGES_REQUESTED** [review]

**Review by v-excelsior — CHANGES_REQUESTED** [review]

File: components/bl-treemap-component/src/index.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      className={ cn("bl-customComponent-treemap", classList, { "bl-customComponent-treemap--disabled": disabled }) }>
+      <IgrTreemap
+        ref={ treemapRef }
+        height="100%"
+        width="100%"
+        rootTitle={ label }
+        valueMemberPath="value"
+        parentIdMemberPath="parent"
+        labelMemberPath="label"
+        idMemberPath="label"
```
[the remark this answers is not in the document]
Comment (Saharij) [reply components/bl-treemap-component/src/index.js:40]: Actually, we don't need this attribute at all.

File: components/bl-treemap-component/src/index.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
+import { useRef } from 'react';
+
+import { useStyles } from './use-styles';
+
+import { IgrTreemap } from './lib/igr-treemap.umd.js';
```
Comment (Valodya) [inline components/bl-treemap-component/src/index.js:5]:
```
import { IgrTreemap, IgrTreemapModule } from './lib/igr-treemap.umd.js';
```

**Review by Valodya — APPROVED** [review]

**Pull request merged**